graf2d/feynman/inc/TFeynman.h

```diff
@@ -0,0 +1,156 @@
+ * Copyright (C) 1995-2000, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#include <iostream>
+#include "TMath.h"
+#include "TVector3.h"
+#include <TLine.h>
+
+class TFeynman {
+
+	public:
+
+		// The constructor
+		TFeynman(Double_t canvasWidth, Double_t canvasHeight){
+			TCanvas *c1 = new TCanvas("c1", "c1", 10,10, canvasWidth, canvasHeight);
+   			c1->Range(0, 0, 140, 60);
+			gStyle->SetLineWidth(2);
+		}
+
+		// Fermions:
+		// Quarks:
+        // plots a Quark
+		void Quark(
+            Double_t x1, // x-Coordinate of starting point
+            Double_t y1, // y-coordinate of starting point
+            Double_t x2, // x-coordinate of second point
+            Double_t y2, // y-coordinate of second point
+            Double_t labelPositionX, // x-coordinate of the particle-label
+            Double_t labelPositionY, //y-coordinate of the particle label 
+            const char * quarkName, // name of the quark (either u, d, c, s, t, b or q)
+            bool isMatter // is the particle matter (if false then antimatter)
+        );
+		 // plots a Quark-Antiquark pair in a cirlce
+		void QuarkAntiQuark(
+            Double_t x1, // 
+            Double_t y1, // coordinates of center of the circle
+            Double_t rad, // radius of the circle
+            Double_t labelPositionX, // 
+            Double_t labelPositionY, // coordinates of the label (just define the coordinates of the first label)
+            const char * quarkName // name of the first quark (u, d, c, s, t, b, q), antiparticle is automatically generated
+            );
+                
+		// Leptons:
+        // plots a Lepton
+		void Lepton(Double_t x1, //
+            Double_t y1, // starting coordinates
+            Double_t x2, 
+            Double_t y2, // stopping coordinates
+            Double_t labelPositionX, //
+            Double_t labelPositionY, // label coordinates
+            const char * whichLepton, // name of the lepton (e, en, m, mn, t, tn) -> Electron, Electron-Neutrino etc.
+            bool isMatter // true or false if it is matter or antimatter
+            );
+		// plots a Lepton-Antilepton pair in a cirlce
+		void LeptonAntiLepton(Double_t x1, //
+            Double_t y1, // coordinates of center of the circle
+            Double_t rad, // radius of the circle
+            Double_t labelPositionX, //
+            Double_t labelPositionY, // coordinates of the circle
+            const char * whichLepton, // name of the first lepton
+            const char * whichAntiLepton // name of the antilepton);
+        ); 
+		
+		// Bosons: 
+        // Guage Bosons:
```

Comment:
Gauge ?

Reply:
In particle physics, a gauge boson is a force carrier, a bosonic particle that carries any of the fundamental interactions of nature, commonly called forces. Elementary particles, whose interactions are described by a gauge theory, interact with each other by the exchange of gauge bosons—usually as virtual particles. I misspelt it here, sorry 😅